draft-irtf-cfrg-hpke.md

```diff
@@ -510,10 +510,8 @@ The primary differences from the base case are:
 * The PSK ID is added to the context string used as the `info` input
   to the KDF
 
-This mechanism is not suitable for use with a low-entropy password
-as the PSK.  A malicious recipient that does not possess the PSK can
-use decryption of a plaintext as an oracle for performing offline
-dictionary attacks.
+The PSK SHOULD be at least Nh-bytes long, and SHOULD be derived from at least
+Nh bytes of entropy. Low-entropy PSKs are subject to active dictionary attacks.
```

Comment:
Is this actually true?  How would such a dictionary attack be mounted without access to either the ephemeral private key or the recipient private key?

Reply:
Say a given client and server have a low entropy PSK k. The adversary can search for it by sending messages to the server, with test PSKs, and seeing which of those are decrypted successfully or not.

Reply:
Hm, that seems like an oracle provided by a higher-layer protocol, not by HPKE.

Reply:
Is it? (Either way, we should probably say *something* about it, right?) What text would you propose?

Reply:
I think a dictionary attack would only be feasible if the adversary already knows the KEM keys, like the ephemeral and the static key(s). Then the adversary can use the decryption of the authenticated encryption as an oracle: it fails if the key is not correct.

A quantum adversary could for example know all KEM keys of a classic (non-post-quantum) KEM, but not the PSK, and then mount this attack.

However, the statement about low-entropy PSKs depends on the employed KDF algorithm. At this place in the spec, the definitions are still rather general. Only later it becomes obvious that HPKE is specified with HKDF and SHA2. HKDF's RFC clearly states that it is not designed to slow down dictionary attacks [1].

[1] https://hyp.is/xAKxmG4hEeqyB5fD4eRdUQ/tools.ietf.org/html/rfc5869

Reply:
Yes, in the example I gave, the adversary is active and knows the sender ephemeral keys.

Reply:
As discussed with @chris-wood off-channel yesterday, I folded this into pull request #50 and make a proposal for a text there.